app-editors/scite/scite-4.1.2.ebuild

```diff
@@ -0,0 +1,112 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit flag-o-matic gnome2-utils toolchain-funcs xdg-utils
+
+MY_PV=${PV//./}
+DESCRIPTION="A very powerful, highly configurable, small editor with syntax coloring."
+HOMEPAGE="http://www.scintilla.org/SciTE.html"
+SRC_URI="http://www.scintilla.org/${PN}${MY_PV}.tgz"
+
+LICENSE="HPND lua? ( MIT )"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86 ~x86-fbsd ~amd64-linux ~arm-linux ~x86-linux"
+IUSE="lua"
+
+RDEPEND="dev-libs/glib:=
+	x11-libs/cairo
+	x11-libs/gdk-pixbuf
+	x11-libs/gtk+:3=
+	x11-libs/pango
+	lua? ( >=dev-lang/lua-5:= )"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	>=sys-devel/gcc-7"
```

Comment:
This isnt reliable; everyone should have it installed but they can **choose** to use earlier version. The `gcc-major-version` check you have should suffice.